api/types.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package api holds commonly used types, constants, defaults, and support functions
+// that are related to the api client implementation.
+package api
+
+import (
+	"time"
+)
+
+// Duration is a wrapper around duration to set up custom marshal/unmarshal
+type Duration time.Duration
```

Comment:
I wonder if we can just use `gogoproto.casttype=time.Duration` in the `.proto` file?